### PR TITLE
Selected File Extensions

### LINKED ISSUE #590: FileProperty does not store extension/type of selected reader
When a `FileProperty` is serialized, it stores a list of all registered file extensions (`<nameFilter>`). However, neither the currently used reader nor the extension is not stored. This can cause problems when multiple readers are registered for the same file extension.

Example: 
* TIFF-Reader registers `tif` and `tiff
* DICOM-Reader registers `tif` and `dcm`
* TIFF volume data is loaded in `VolumeSource` with `TIFF reader` selected, then the workspace is saved
* Loading the workspace results in an error that the DICOM reader failed to read the tiff file (it might be the first reader in the factory handling a tiff extension)

How should we deal with this?

### PR DESCRIPTION
**Pull request by petersteneteg**
Updates to the FileProperty and Source processors. Better handling of the selected file extensions.

closes #590

bild

### CODE REVIEW

**Comment by petersteneteg** [comment]
review anyone?

File: modules/base/src/processors/volumesequencesource.cpp [hunk 1]
```diff
@@ -167,15 +190,11 @@ void VolumeSequenceSource::loadFolder(bool deserialize) {
...
 }
 
-void VolumeSequenceSource::addFileNameFilters() {
-    file_.clearNameFilters();
-    file_.addNameFilter(FileExtension::all());
-    file_.addNameFilters(rf_->getExtensionsForType<VolumeSequence>());
-}
-
 void VolumeSequenceSource::process() {
     if (file_.isModified() || reload_.isModified() || folder_.isModified() ||
```
Comment (r-englund) [inline modules/base/src/processors/volumesequencesource.cpp:199]: do we need to add reader_.isModified() here?

**Review by r-englund — APPROVED** [review]

File: modules/base/src/processors/volumesequencesource.cpp (hunk 1 above)
Comment (petersteneteg) [reply modules/base/src/processors/volumesequencesource.cpp:199]: yes

File: include/inviwo/core/properties/fileproperty.h [hunk 2]
```diff
@@ -82,7 +82,14 @@ class IVW_CORE_API FileProperty : public TemplateProperty<std::string> {
     virtual FileProperty* clone() const override;
     virtual ~FileProperty() = default;
 
-    virtual void set(const std::string& value) override;
+    /**
+     * Set the file name and also update the selected extension to the first one matching file.
```
Comment (ResearchDaniel) [inline include/inviwo/core/properties/fileproperty.h:86]: first one matching file? 
You mean?
Set file name and update the selected extension to that of the file.

File: include/inviwo/core/properties/fileproperty.h [hunk 3]
```diff
@@ -101,7 +108,7 @@ class IVW_CORE_API FileProperty : public TemplateProperty<std::string> {
     FileMode getFileMode() const;
 
     void setContentType(const std::string& contentType);
-    std::string getContentType() const;
+    const std::string& getContentType() const;
```
Comment (ResearchDaniel) [inline include/inviwo/core/properties/fileproperty.h:111]: not part of this PR but a hint on what content type is would be good...

**Review by ResearchDaniel — APPROVED** [review]
Looks good. I guess we could hide the reader_ if there is just a single reader?

File: include/inviwo/core/properties/fileproperty.h (hunk 2 above)
Comment (petersteneteg) [reply include/inviwo/core/properties/fileproperty.h:86]: the file is just a string, and there can be may FileExtensions that match for example *.jpg (one per registered reader)
